wormhole/file_transport.go

```diff
@@ -460,44 +538,77 @@ func (t *fileTransport) listen() error {
 }
 
 func (t *fileTransport) listenRelay() error {
+	fmt.Println("hello from listenRelay")
+
+	// TODO: consider if/how contexts are being used and be consistent and/or correct.
+	ctx := context.Background()
 	if t.relayAddr == "" {
 		return nil
 	}
-	conn, err := net.Dial("tcp", t.relayAddr)
-	if err != nil {
-		return err
+
+	hostParts := strings.SplitN(t.relayAddr, "://", 2)
+	fmt.Printf("hostParts: %v\n", hostParts)
+	var (
+		conn net.Conn
+		err  error
+	)
+	if len(hostParts) == 2 {
+		fmt.Println("hostParts == 2")
+		proto := hostParts[0]
+		if proto == "ws" || proto == "wss" {
+			fmt.Println("proto == ws/wss")
+			wsConn, _, err := ws.Dial(ctx, t.relayAddr, nil)
+			if err != nil {
+				return err
+			}
+			fmt.Println("conn = ws.NetConn")
+			conn = ws.NetConn(ctx, wsConn, ws.MessageBinary)
+		}
+	} else {
+		fmt.Println("conn = net.Dial tcp")
+		conn, err = net.Dial("tcp", t.relayAddr)
+		if err != nil {
+			return err
+		}
 	}
 
+	fmt.Println("connWrite")
 	_, err = conn.Write(t.relayHandshakeHeader())
 	if err != nil {
 		conn.Close()
 		return err
 	}
 
+	fmt.Println("t.relayConn = conn")
 	t.relayConn = conn
 	return nil
 }
 
 func (t *fileTransport) waitForRelayPeer(conn net.Conn, cancelCh chan struct{}) error {
+	fmt.Println("hello from waitForRelayPeer")
 	okCh := make(chan struct{})
 	go func() {
 		select {
 		case <-cancelCh:
+			fmt.Println("<-cancelCh")
 			conn.Close()
 		case <-okCh:
+			fmt.Println("<-okCh")
 		}
 	}()
 
 	defer close(okCh)
 
-	gotOk := make([]byte, 3)
+	gotOk := make([]byte, 2)
 	_, err := io.ReadFull(conn, gotOk)
 	if err != nil {
+		fmt.Printf("waitForRelayPeer->io.ReadFull err: %s\n", err)
 		conn.Close()
 		return err
 	}
 
-	if !bytes.Equal(gotOk, []byte("ok\n")) {
+	fmt.Printf("gotOk: %v", gotOk)
+	if !bytes.Equal(gotOk, []byte("ok")) {
 		conn.Close()
 		return errors.New("got non ok status from relay server")
 	}
```

Comment:
OK, looks good. Guess many of these debug prints would go away in a future commit.

Reply:
Yeah, the print statements need to be removed. I think for spiking we don't want to be merging anyways. I think the next step here would be to start fresh from master and test-drive out the behavior we need. I suspect some combination of cherry-picking, checkout, amending, and/or copy/paste may be the best way to export the useful bits out of here and into the test, leaving the garbage behind.